hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/HoodieSqlCommonUtils.scala

```diff
@@ -223,6 +223,17 @@ object HoodieSqlCommonUtils extends SparkAdapterSupport {
   def isHoodieConfigKey(key: String): Boolean =
     key.startsWith("hoodie.") || key == DataSourceReadOptions.TIME_TRAVEL_AS_OF_INSTANT.key
 
+  /**
+   * Extract hoodie config from conf using prefix "spark.hoodie." and "hoodie.".
+   */
+  def extractHoodieConfig(conf: Map[String, String]): Map[String, String] = {
```

Comment:
Should `ProvidesHoodieConfig#combineOptions` call this new method to cover the code path of `Command`s (`AlterHoodieTableDropPartitionCommand`, `InsertIntoHoodieTableCommand` ..), too?

Reply:
Nice catch, it should be.